android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java

```diff
@@ -190,17 +204,29 @@ private String getClientHash(){
     DynamicRealm getDynamicRealm(){
         synchronized (LOCK){
             Uri server = Uri.parse(client.getBaseUrl());
-            DynamicRealm realm = DynamicRealm.getInstance(getRealmConfiguration());
+            DynamicRealm realm;
+            try {
+                realm = DynamicRealm.getInstance(getRealmConfiguration());
+            } catch (RealmFileException exception) {
+                throw new KinveyException("Access Error", "Use correct encryption key", "You are using wrong encryption key");
```

Comment:
Is it possible to catch another `RealmFileException` that is not an error with encryption key? If so, do we need to inspect the exception before reporting as an encryption key exception?

Reply:
I've seen only this exception, but we can check the RealmFileException's message. If RealmFileException's message has "Access Error" then we will throw KinveyException("Access Error"), in other case the RealmFileException will be thraown.